scripts/swupdate/merge.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/sh
+
+#Below function will return value corresponding to requested key
+function getProperty {
+   PROP_KEY=$1
+   PROPERTY_FILE=$2
+   FILE_TYPE=$3
+   if [ "$fileType" = "properties" ]
+   then
+       PROP_VALUE=`cat $PROPERTY_FILE | grep "$PROP_KEY=" | cut -d'=' -f2`
+   else
+       PROP_VALUE=`cat $PROPERTY_FILE | grep "$PROP_KEY:" | cut -d':' -f2`
+   fi
+   IFS=$'#'
+   read -ra prop_val <<< "$PROP_VALUE"
+   echo "${prop_val[0]// /}"
+}
+
+#Below method perform merging of config
+function performMerge {
+while read line; do
+if [ "$fileType" = "properties" ]
+then
+    IFS=$'='
+else
+    IFS=$':'
+fi
+read -ra ID <<< "$line"
+key="${ID[0]// /}" #remove leading and trailing spaces
+value="${ID[1]// /}"
+if ! grep -Fx "${key}" $unsafeAttributesFile >/dev/null
+then
+if [[ $key != \#* ]]; then #Avoiding commented lines or keys
+configValue=$(getProperty "$key" "$configFile" "$fileType")
+oldSampleValue=$(getProperty "$key" "$oldSampleFile" "$fileType")
+newSampleValue=$(getProperty "$key" "$newSampleFile" "$fileType")
+#if config and old sample value same or new parameter added then replace config value with new sample value
+#else keep config value as it is
+if [ "$configValue" = "$oldSampleValue" ] || [ -z "$configValue" ]
+then
+IFS=$'\n'
+if [ "$fileType" = "properties" ]
+then
+    sed -i "s+$key=$configValue+$key=$newSampleValue+" $configFile
+else
+    sed -i "s+$key: $configValue+$key: $newSampleValue+" $configFile
```

Comment:
This format only mentions 1 space between "key: value".  If ever anyone puts two or more spaces, this replace will silently fail.  I suggest to modify this to support arbitrary number of spaces.

Reply:
Yes but I am not sure how to handle this. Need your help only for this.

Reply:
This should work, checked locally:

```
sed -i "s+$key: *$configValue+$key: $newSampleValue+" $configFile
```

(Added asterics `*` after space -- `key: *`.)

Reply:
Yes its working @t7ko-seagate Have made smal change for same in getProperty-grep command too.
pls check 1s

Reply:
thanks, approved.